protocol/messenger.go

```diff
@@ -4692,54 +4720,86 @@ func (m *Messenger) MarkAllReadInCommunity(communityID string) ([]string, error)
 
 // MuteChat signals to the messenger that we don't want to be notified
 // on new messages from this chat
-func (m *Messenger) MuteChat(chatID string) error {
-	chat, ok := m.allChats.Load(chatID)
+func (m *Messenger) MuteChat(MuteParams *requests.MuteParams) (int64, error) {
+	chat, ok := m.allChats.Load(MuteParams.ChatID)
 	if !ok {
 		// Only one to one chan be muted when it's not in the database
-		publicKey, err := common.HexToPubkey(chatID)
+		publicKey, err := common.HexToPubkey(MuteParams.ChatID)
 		if err != nil {
-			return err
+			return 0, err
 		}
 
 		// Create a one to one chat and set active to false
-		chat = CreateOneToOneChat(chatID, publicKey, m.getTimesource())
+		chat = CreateOneToOneChat(MuteParams.ChatID, publicKey, m.getTimesource())
 		chat.Active = false
 		err = m.initChatSyncFields(chat)
 		if err != nil {
-			return err
+			return 0, err
 		}
 		err = m.saveChat(chat)
 		if err != nil {
-			return err
+			return 0, err
 		}
 	}
 
 	var contact *Contact
 	if chat.OneToOne() {
-		contact, _ = m.allContacts.Load(chatID)
+		contact, _ = m.allContacts.Load(MuteParams.ChatID)
+	}
+
+	switch MuteParams.MutedType {
+	case MuteTill1MinType:
+		MuteParams.MutedTill = time.Now().Unix() + MuteFor1MinDuration
```

Comment:
not sure you need `MutedTill` in the `MuteParams` structure, I would just create a local variable for this, but I might be missing something